### ISSUE TITLE
Test suite fails inside docker containers

### ISSUE THREAD

**Issue by jpsalvesen** [mschout/perl-log-dispatch-filerotate#5]
I get failures when attempting to build Log::Dispatch::FileRotate inside docker containers. 

```
[user@localhost ~]$ docker run -ti perl bash
Unable to find image 'perl:latest' locally
latest: Pulling from library/perl
Digest: sha256:d96aadcdc209340251394f4c2a7bf995cf9868b5bca92902f04bc2349dbefa1f
Status: Downloaded newer image for perl:latest
root@e867f68e3183:~# cpanm Log::Dispatch::FileRotate
...
Building and testing Log-Dispatch-FileRotate-1.24 ... FAIL
! Installing Log::Dispatch::FileRotate failed. See /root/.cpanm/work/1491830460.7/build.log for details. Retry with --force to force install it.
```

When I inspect the build log, I find this:

```
Building and testing Log-Dispatch-FileRotate-1.24
cp lib/Log/Dispatch/FileRotate.pm blib/lib/Log/Dispatch/FileRotate.pm
PERL_DL_NONLAZY=1 "/usr/local/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/author-pod-syntax.t ...... skipped: these tests are for testing by the author
t/basic.t .................. ok
t/file-open-failure.t ...... skipped: locale en_US.UTF-8 is not available on this system

#   Failed test at t/lockfile-open-failure.t line 41.
# didn't find a warning
# expected to find warning: (?^:\d+ Log::Dispatch::FileRotate failed to get lock)

#   Failed test at t/lockfile-open-failure.t line 45.
#          got: 'write with successful lockWrite with unsuccessful lock'
#     expected: 'write with successful lock'
# Looks like you failed 2 tests of 6.
t/lockfile-open-failure.t .. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/6 subtests 
t/no-activity-bug.t ........ ok
t/signature.t .............. skipped: Set the environment variable TEST_SIGNATURE to enable this test.
t/size-with-underscore.t ... ok

Test Summary Report
-------------------
t/lockfile-open-failure.t (Wstat: 512 Tests: 6 Failed: 2)
  Failed tests:  5-6
  Non-zero exit status: 2
Files=7, Tests=245,  4 wallclock secs ( 0.10 usr  0.02 sys +  3.33 cusr  0.37 csys =  3.82 CPU)
Result: FAIL
Failed 1/7 test programs. 2/245 subtests failed.
Makefile:867: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
-> FAIL Installing Log::Dispatch::FileRotate failed. See /root/.cpanm/work/1491830407.7/build.log for details. Retry with --force to force install it.
40 distributions installed

```

I get the same failures with perl 5.14.4 plus older moduls as 5.24.1 with the latest modules.

**Comment by mschout** [comment]
I took a quick look at this and I think what is going on here is that because the user running the tests in the perl docker container is "root", the lock is obtained even though the lock file permissions are 000.   I can reproduce this behavior outside docker as well.  I'll come up with something else (perhaps externally flock()'ing the file in the test file) or, we'll have to skip this test if running under UID 0

**Comment by mschout** [comment]
t/file-open-failure.t would also fail if locale en_US.UTF-8 was available...

root user is exempt from access permissions tests (execept execute bit).  Probably the best we can do is skip these tests if the user running the test suite is root.

**Comment by mschout** [comment]
Fixed (by skipping) in v1.25, which has just been uploaded to CPAN
